source/plugin/scheduler.rst

```diff
@@ -163,3 +163,149 @@ In addition, there are a few other operations that are safe to do asynchronously
 * Filesystem I/O (excluding files used by Sponge)
+
+Compatibility with other libraries
+==================================
+
+As your plugin grows in size and scope you might want to start using one of the many concurrency libraries available 
+for Java and the JVM.
+These libraries do tend to support Java's 
+`ExecutorService <https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ExecutorService.html>` as a means 
+of directing on which thread the task is executed.
+
+To allow these libraries to work with Sponge's ``Scheduler`` the following methods can be used:
+
+* ``Scheduler#createSyncExecutor(Object plugin)`` creates a ``ScheduledExecutorService`` which executes tasks
+  through Sponge's synchronous scheduler.
+* ``Scheduler#createAsyncExecutor(Object plugin)`` creates a ``ScheduledExecutorService`` which executes tasks
+  through Sponge's asynchronous scheduler. Tasks are subject to the restrictions mentioned in `Asynchronous Tasks`_.
+
+One thing to keep in mind is that any tasks that interacts with Sponge outside of the interactions listed in 
+`Asynchronous Tasks`_ need to be executed on the ExecutorService created with 
+``Scheduler#createSyncExecutor(Object plugin)`` to be thread-safe.
+
+.. code-block:: java
+
+    import org.spongepowered.api.scheduler.SpongeExecutorService;
+    
+    SpongeExecutorService minecraftExecutor = Sponge.getScheduler().createSyncExecutor(plugin);
+    
+    minecraftExecutor.submit(() -> { ... });
+    
+    minecraftExecutor.schedule(() -> { ... }, 10, TimeUnit.SECONDS);
+
+Almost all libraries have some way of adapting the ``ExecutorService`` to natively schedule tasks.
+As an example the following paragraphs will explain how the ``ExecutorService`` is used in a number of libraries.
+
+CompletableFuture (Java 8)
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. _CompletableFuture: https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/CompletableFuture.html
+
+With Java 8 the CompletableFuture_ object was added to the standard library.
+Compared to the ``Future`` object, this allows for the developer to provide a callback that is called when the future
+completes rather than blocking the thread until the future eventually completes.
+
+CompletableFuture_ is a fluent interface which usually has the following three variations for each of its functions:
+
+* ``CompletableFuture#<function>Async(..., Executor ex)`` Executes this function through ``ex``
+* ``CompletableFuture#<function>Async(...)`` Executes this function through ``ForkJoinPool.commonPool()``
+* ``CompletableFuture#<function>(...)`` Executes this function on whatever thread the previous ``CompletableFuture`` was completed on.
+
+.. code-block:: java
+
+    import java.util.concurrent.CompletableFuture;
+    
+    SpongeExecutorService minecraftExecutor = Sponge.getScheduler().createSyncExecutor(plugin);
+
+    CompletableFuture.supplyAsync(() -> {
+        // ASYNC: ForkJoinPool.commonPool()
+        return 42
```

Comment:
@12AwsomeMan34 Thats what I get for working in 4 different JVM languages at once...